src/main.rs

```diff
@@ -75,53 +63,35 @@ struct Newer {
 
 impl Newer {
     fn new(args: Vec<String>) -> Result<Newer, Error> {
-        enum State {
-            Output,
-            Input,
-            Command,
-        }
+        let mut args = args.iter();
+
+        let newest = args
+            .by_ref()
+            .take_while(|&a| a != ":")
+            .map(|arg| match find_newest(arg) {
+                Err(e) if e.kind() == ErrorKind::NotFound => Ok(SystemTime::UNIX_EPOCH),
+                n @ _ => n,
+            })
```

Comment:
nit: a `.filter_map` is probably more appropriate here. Something like:

```rust
.filter_map(|arg| match find_newest(arg) {
    Err(e) if e.kind() == ErrorKind::NotFound => None,
    n @ _ => Some(n),
})
```

Reply:
Definitely not a fan of using iterators for this, though it's interesting.